pyquil/quil.py

```diff
@@ -40,6 +40,9 @@ def synthesize(self, resource_manager=None):
         self.resource_manager.reset()
         return super(Program, self).synthesize(resource_manager)
 
+    def __eq__(self, other):
```

Comment:
This will be brittle when there are managed qubits. I recommend making this slightly more robust to generated qubit names. Pseudocode:

```
reset_numbers()
x = self.out()
reset_numbers()
y = other.out()
return x == y
```

The idea of using `==` anyway to me doesn't seem like a good idea. I'd prefer something like a function called `same_instructions`, that's a little more mnemonic.